src/iodide-api/evalQueue.js

```diff
@@ -34,7 +59,29 @@ export const waitForExplicitContinuationStatusResolution = () => new Promise((re
   }
 })
 
+const awaitPromises = promises =>
+  // is functionally identical to Promise.all.
+  Promise.resolve()
```

Comment:
Is there a reason why you're creating a problem that is immediately resolved here? Could we not just key this off of Promises.all() and do whatever state setting we want to do immediately?

Reply:
yep this is madness, fixing